static/app/components/searchQueryBuilder/searchQueryBuilderContext.tsx

```diff
@@ -0,0 +1,25 @@
+import {createContext, type Dispatch, useContext} from 'react';
+
+import type {QueryBuilderActions} from 'sentry/components/searchQueryBuilder/useQueryBuilderState';
+import type {ParseResult} from 'sentry/components/searchSyntax/parser';
+import type {Tag, TagCollection} from 'sentry/types';
+
+interface SearchQueryBuilerContext {
+  dispatch: Dispatch<QueryBuilderActions>;
+  getTagValues: (tag: Tag, query: string) => Promise<string[]>;
+  keys: TagCollection;
+  parsedQuery: ParseResult | null;
+  query: string;
+}
+
+export function useSearchQueryBuilder() {
+  return useContext(SearchQueryBuilerContext);
+}
+
+export const SearchQueryBuilerContext = createContext<SearchQueryBuilerContext>({
```

Comment:
Can you do this even though the variable and type has the same name lol

Reply:
Good question, I guess you can hah. Changed the type so their not the same.